src/main/resources/Diagram/MigrationScript.xml

```diff
@@ -0,0 +1,144 @@
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+-->
+
+<xwikidoc version="1.3" reference="Diagram.MigrationScript" locale="">
+  <web>Diagram</web>
+  <name>MigrationScript</name>
+  <language/>
+  <defaultLanguage/>
+  <translation>0</translation>
+  <creator>xwiki:XWiki.Admin</creator>
+  <creationDate>1549541592000</creationDate>
+  <parent>xwiki:Diagram.WebHome</parent>
+  <author>xwiki:XWiki.Admin</author>
+  <contentAuthor>xwiki:XWiki.Admin</contentAuthor>
+  <date>1549541642000</date>
+  <contentUpdateDate>1549541592000</contentUpdateDate>
+  <version>1.1</version>
+  <title>MigrationScript</title>
+  <comment/>
+  <minorEdit>false</minorEdit>
+  <syntaxId>xwiki/2.1</syntaxId>
+  <hidden>true</hidden>
+  <content>{{info}}
+This page contains a script for migrating diagram pages after an upgrade of the diagram application to the pro version of the application. It will update the paths containing the version of the draw.io library with the version currently used by the application.
+
+{{html}}
+  &lt;form action="" method="post"&gt;
+  &lt;input class="btn btn-default" type="submit" name="preview" value="Preview"/&gt;
+  &lt;input class="btn btn-default" type="submit" name="execute" value="Execute"/&gt;
+  &lt;/form&gt;
+{{/html}}
+
+{{/info}}
+
+{{groovy}}
+  
+
+String getDependencyVersion(String extensionId, String dependencyId) {
+
+  def extensionManager = services.extension
+  def repository = extensionManager.getRepository('installed')
+
+  def extensions = services.extension.installed.getInstalledExtensions()
+  for (extension in extensions) {
+    if ("$extension.id".indexOf(extensionId) == 0) {
+      for (dependency in extension.dependencies) {
+        if (dependency.id.indexOf(dependencyId) == 0) {
+          return dependency.versionConstraint
+        }
+      }
+    }
+  }
+  return "No version found, make sure the extension $extensionId is installed."
+}
+
+if (request.preview || request.execute) {
+  def targetVersion = getDependencyVersion("com.xwiki.diagram:application-diagram", "org.xwiki.contrib:draw.io")
+
+  println "The target draw.io version found is : $targetVersion."
+
+  def hql = "select doc.fullName from XWikiDocument as doc, BaseObject as obj where obj.name = doc.fullName and obj.className='Diagram.DiagramClass' and doc.fullName &lt;&gt; 'Diagram.DiagramTemplate'"

```

Comment:
Can't we retrieve only the diagrams that need to be migrated? I'm asking also because it would be nice to show a message on the Diagram home page when there are diagrams that need to be migrated (e.g. "You have X diagrams that need to be migrated"). Also, if we could simply filter ``doc.content`` then it would be relatively easy to write a live table that shows the diagrams that need to be migrated (``LiveTableResultsMacros`` supports an extra where). And then the message from the home page would have a link to this live table.

Reply:
This sounds good indeed, but I can't figure out so far how to pass a parameter relating to "doc.content" to a livetable. I assume that you are referring to something else than the "extraParams" parameter, right (since that one only supports the "space", "orphaned", "parent" properties plus the ones relating to specific classes)? Are you suggesting to create a custom livetable results page (just checking before following this path)?

Reply:
I actually found my way into this filtering mechanism thanks to @lucaa, I'm finalizing the update.